framework/filters/AccessRule.php

```diff
@@ -136,19 +142,26 @@ protected function matchController($controller)
      */
     protected function matchRole($user)
     {
-        if (empty($this->roles)) {
+        $privileges = empty($this->roles) ? [] : $this->roles;
+
```

Comment:
the introduction of the new 'variable' `$privileges` and the fact that it is assembled by a merge of $roles & $permissions is a bit too magical IMO. 

Ideally, this should be replaced with an OO approach